Mathlib/Analysis/Discrete/MarcinkiewiczZygmund.lean

```diff
@@ -0,0 +1,290 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Algebra.Order.Chebyshev
+import Mathlib.Analysis.InnerProductSpace.PiL2
+import Mathlib.Data.Nat.Choose.Multinomial
+import Mathlib.Tactic.Positivity.Finset
+
+/-!
+# The Marcinkiewicz-Zygmund inequality
+
+This file proves the Marcinkiewicz-Zygmund inequality. This is the statement that, for a random
+variable `f` valued in a real inner product space, the $$L^p$$-norm of the sum of `n` samples of `f`
+is bounded by the $$L^p$$-norm of `f` up to some multiplicative constant.
```

Comment:
I am not sure I understand this quibble. Isn't the point to get *any* constant? At least, [Wikipedia](https://en.wikipedia.org/wiki/Marcinkiewicz%E2%80%93Zygmund_inequality) only claims the result with a constant, without specifying anything about its size and relationship to the central limit theorem.

Reply:
I think the quibble here is exactly the opposite! The usual MZ inequality allows any constant, but in our specific application the shape of the constant is actually important. And so what LeanAPAP needs as "the MZ inequality" is a bit different from the usual formulation, since we require better control on the constants. And then the docstring should say a bit more about what the control needed is

Reply:
(imo, your new docstring satisfies this)